cmd/protoc-gen-goclay/genhandler/handler.go

```diff
@@ -346,12 +346,24 @@ func (g *Generator) getServiceImpl(f *descriptor.File, s *descriptor.Service) (s
 	return applyImplTemplate(g.getImplParam(f, s, nil, []string{"github.com/utrack/clay/v2/transport"}))
 }
 
-func (g *Generator) getMethodImpl(f *descriptor.File, s *descriptor.Service, m *descriptor.Method) (string, error) {
-	return applyImplTemplate(g.getImplParam(f, s, m, []string{"context", "github.com/pkg/errors"}))
+func (g *Generator) getMethodImpl(s *descriptor.Service, m *descriptor.Method) (string, error) {
+	// restore orig GoPkg
+	savedPkg := m.RequestType.File.GoPkg
+	defer func() {
+		m.RequestType.File.GoPkg = savedPkg
+	}()
+
+	return applyImplTemplate(g.getImplParam(m.RequestType.File, s, m, []string{"context", "github.com/pkg/errors"}))
 }
 
-func (g *Generator) getTestImpl(f *descriptor.File, s *descriptor.Service, m *descriptor.Method) (string, error) {
-	return applyTestTemplate(g.getImplParam(f, s, m, []string{"context", "testing", "github.com/stretchr/testify/require"}))
+func (g *Generator) getTestImpl(s *descriptor.Service, m *descriptor.Method) (string, error) {
+	// restore orig GoPkg
+	savedPkg := m.RequestType.File.GoPkg
```

Comment:
Are you sure that all these vars in the chain `m.RequestType.File.GoPkg` are being set in all the cases?

Reply:
Yes, at the moment I'm sure, but I can leave all of params and not to use that chain of a method, as you wish. I think, this part of code needs more complex refactoring, so sometimes it will be reorganized

Reply:
I was just passing by.
It's up to you, guys.
But maybe it was there for a reason.

Reply:
This chain being used only inside two functions, each of them is for some (exactly) method generation, so them have to contain  *descriptor.Method and as you can see inside handler.go at 393/394 - it refers to m.RequestType.File.GoPkg.Path / m.ResponseType.File.GoPkg.Path without checking all part of the chain

Reply:
`But maybe it was there for a reason`
I don't think so, but I'm waiting for Nick's comment. I think it was there because of time flow %) As this method was implemented for service and for method generation with similar semantics...

Reply:
A-ha. Yes, everything in that chain must be set (assuming `m` is not nil), unless there's something seriously wrong with the registry. 

I've tried to move to another registry and get rid of hacks like these in #78 - but it seems to be too unstable to do so right away, since we depend a lot on grpc-gateway internals and proto options that mutate them. 

I plan to abstract final templates from grpc-gateway internal functions (and refactor [this mapfunc mess](https://github.com/utrack/clay/blob/93af5eb45b028f560a542ea141f2eb2e4c459551/cmd/protoc-gen-goclay/genhandler/template.go#L218) in the process) first.

Reply:
It's a great news!